### PR TITLE
feat: step header button

### PR DESCRIPTION
**Pull request by levi-zustiak**
# Description

### Changes
- Adds the `StepHeaderMenu` component

### CODE REVIEW

File: libs/journeys/ui/src/components/StepHeader/StepHeader.tsx [hunk 1]
```diff
@@ -46,7 +48,7 @@ export function StepHeader({
           justifyContent="space-between"
           spacing={2}
           sx={{
-            px: { xs: 6, lg: 0 },
+            px: { xs: 4, lg: 0 },
```
Comment (csiyang) [inline libs/journeys/ui/src/components/StepHeader/StepHeader.tsx:51]: I ended up using 3 here to make sure the logo alignment was correct, will this work for you too?
```suggestion
            px: { xs: 3, lg: 0 },
```

File: libs/journeys/ui/src/components/StepHeader/StepHeader.tsx [hunk 2]
```diff
@@ -84,7 +86,9 @@ export function StepHeader({
                 {journey?.displayTitle ?? journey?.seoTitle}
               </Typography>
             </Stack>
+
             {/* Menu */}
+            <StepHeaderMenu />
```
Comment (csiyang) [inline libs/journeys/ui/src/components/StepHeader/StepHeader.tsx:91]: Remove comment and space
```suggestion
            <StepHeaderMenu />
```

File: libs/journeys/ui/src/components/StepHeader/StepHeaderMenu/StepHeaderMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import { useJourney } from '../../../libs/JourneyProvider'
+import { transformer } from '../../../libs/transformer'
+import { getMenuIcon } from '../utils/getMenuIcon'
+
+export function StepHeaderMenu(): ReactElement {
+  const { journey, variant } = useJourney()
+
+  const router = useRouter()
+
+  const blocks = transformer(journey?.blocks ?? [])
```
Comment (csiyang) [inline libs/journeys/ui/src/components/StepHeader/StepHeaderMenu/StepHeaderMenu.tsx:19]: any reason we need to look through all the blocks. We can just use `journey.menuStepBlock`?

**Review by csiyang — CHANGES_REQUESTED** [review]

File: libs/journeys/ui/src/components/StepHeader/StepHeader.tsx (hunk 1 above)
Comment (levi-zustiak) [reply libs/journeys/ui/src/components/StepHeader/StepHeader.tsx:51]: Yeah, I don't see any issues with that

Comment (csiyang) [reply libs/journeys/ui/src/components/StepHeader/StepHeader.tsx:51]: Great, can you make this change then?

File: libs/journeys/ui/src/components/StepHeader/StepHeaderMenu/StepHeaderMenu.tsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    <Box
+      sx={{
+        borderRadius: 100,
+        border: menuButtonIcon == null && variant === 'admin' ? 'dashed' : null,
+        minHeight: 48,
+        minWidth: 48,
+        display: 'grid',
+        placeItems: 'center',
+        maxWidth: 'min-content',
+        opacity: menuButtonIcon == null ? 0.6 : 1
```
Comment (csiyang) [inline libs/journeys/ui/src/components/StepHeader/StepHeaderMenu/StepHeaderMenu.tsx:64]: To match the other placeholder buttons
```suggestion
        opacity: menuButtonIcon == null ? 0.5 : 1
```

**Review by csiyang — CHANGES_REQUESTED** [review]

File: libs/journeys/ui/src/components/StepHeader/StepHeader.tsx (hunk 1 above)
Comment (levi-zustiak) [reply libs/journeys/ui/src/components/StepHeader/StepHeader.tsx]: Yeah I forgot to add it.. my bad haha

**Review by csiyang — APPROVED** [review]

**Review by csiyang — DISMISSED** [review]

**Review by csiyang — APPROVED** [review]

**Pull request merged**